tests/lib/rules/no-settled-after-test-helper.js

```diff
@@ -0,0 +1,145 @@
+const RuleTester = require('eslint').RuleTester;
+const rule = require('../../../lib/rules/no-settled-after-test-helper');
+
+const { ERROR_MESSAGE } = rule;
+
+const ruleTester = new RuleTester({
+  parser: require.resolve('babel-eslint'),
+  parserOptions: {
+    ecmaVersion: 2015,
+    sourceType: 'module',
+  },
+});
+
+ruleTester.run('no-settled-after-test-helper', rule, {
+  valid: [
+    `
+      import { waitFor, settled } from '@ember/test-helpers';
+
+      async () => {
+        await waitFor('.foo');
+        await settled();
+      }
+    `,
+    `
+      import { waitFor, settled } from '@ember/test-helpers';
+
+      async () => {
+        await waitFor('.foo');
+        await settled;
+      }
+    `,
+    `
+      import { settled } from '@ember/test-helpers';
+
+      async () => {
+        await settled();
+      }
+    `,
+    `
+      import { settled } from '@ember/test-helpers';
+
+      async () => {
+        this.set('foo', 42);
+        await settled();
+      }
+    `,
+    `
+      import { settled } from '@ember/test-helpers';
+
+      async () => {
+        let promise = waitFor('.foo');
+        await promise;
+        await settled();
+      }
+    `,
+    `
+      import { settled } from '@ember/test-helpers';
+
+      function click() {}
+
+      async () => {
+        await click('.foo');
+        await settled();
+      }
+    `,
+  ],
+  invalid: [
```

Comment:
Can we catch things without `await`? I know `await` is supposed to be used with this helper but we might as well catch instances that are missing `await` too.

```js
import { settled } from '@ember/test-helpers';

settled();
settled();
```

Reply:
I'm not sure what to catch there? We might consider a different rule that warns about using test helpers without 1) assigning the resulting promise to a variable, 2) using e.g. `.then()` on them or 3) using `await` with them. I don't think we should put that stuff in this rule though.

Reply:
I agree we should have another rule to ensure the async helpers are called in a way that waits for the promise. But I don't fully understand why the current rule shouldn't catch settled after a test helper when used without await like this:

```js
import { fillIn, settled } from '@ember/test-helpers';

fillIn('.foo'); 
settled();
```

Reply:
because that code does not do anything on it's own if nothing is waiting for the promise that the helper returns.